refl1d/errors.py

```diff
@@ -357,34 +385,36 @@ def _write_file(path, data, title, columns):
 def dark(color):
     return dhsv(color, dv=-0.2)
 
-def _profiles_overplot(profiles):
+def _profiles_overplot(profiles, axes=None):
     for model, group in profiles.items():
         name = model.name
         absorbing = any((L[2] != 1e-4).any() for L in group)
         magnetic = (len(group[0]) > 3)
         # Note: Use 3 colours per dataset for consistency
-        _draw_overplot(group, 1, name + ' rho')
+        _draw_overplot(group, 1, name + ' rho', axes=axes)
         if absorbing:
-            _draw_overplot(group, 2, name + ' irho')
+            _draw_overplot(group, 2, name + ' irho', axes=axes)
         else:
-            next_color()
+            next_color(axes=axes)
         if magnetic:
-            _draw_overplot(group, 3, name + ' rhoM')
+            _draw_overplot(group, 3, name + ' rhoM', axes=axes)
         else:
-            next_color()
-    _profile_labels()
+            next_color(axes=axes)
+    _profile_labels(axes=axes)
 
-def _draw_overplot(group, index, label):
+def _draw_overplot(group, index, label, axes=None):
```

Comment:
Passing fig and axes around everywhere (here and in bumps) is tedious.

Reply:
Yes!  and also thread-safe.